nameko/amqp.py

```diff
@@ -47,3 +51,33 @@ def verify_amqp_uri(amqp_uri):
     transport = TestTransport(connection.transport.client)
     with transport.establish_connection():
         pass
+
+
+class UndeliverableMessage(Exception):
+    """ Raised when publisher confirms are enabled and a message could not
+    be routed or persisted """
+    pass
+
+
+@contextmanager
+def get_connection(amqp_uri):
+    conn = Connection(amqp_uri)
+    with connections[conn].acquire(block=True) as connection:
+        yield connection
+
+
+@contextmanager
+def get_producer(amqp_uri, confirms=True):
+    transport_options = {
+        'confirm_publish': confirms
+    }
+    conn = Connection(amqp_uri, transport_options=transport_options)
+
+    with producers[conn].acquire(block=True) as producer:
+        yield producer
+        try:
+            returned = producer.channel.returned_messages.get_nowait()
+        except Empty:
+            pass
+        else:
+            raise UndeliverableMessage(returned)
```

Comment:
i don't think the event dispatcher wants to raise here. there might not be anyone subscribed to the events yet. we use the confirm to get the server to "acknowledge" our message, but returned messages are also fine in this case